tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -1197,6 +1245,32 @@ std::vector<Tensor> Matmul::create_output_tensors(const std::vector<Tensor>& inp
                         output_layout,
                         input_tensor_a.device(),
                         mem_config)};
+                } else if constexpr (std::is_same_v<
+                                         ProgramConfigType,
+                                         MatmulMultiCoreReuseMultiCastDRAMShardedProgramConfig>) {
+                    uint32_t M =
+                        (program_config.fuse_batch ? input_tensor_a.volume() / input_tensor_a.get_legacy_shape()[-1]
+                                                   : input_tensor_a.get_legacy_shape()[-2]) /
+                        TILE_HEIGHT;
+                    uint32_t N = input_tensor_b.get_legacy_shape()[-1] / TILE_WIDTH;
+                    auto input_tensor_b_shape = input_tensor_b.get_legacy_shape();
+                    uint32_t N_unpaddded = input_tensor_b.get_legacy_shape()[-1] -
+                                           input_tensor_b_shape.padding()[input_tensor_b_shape.rank() - 1].back;
+
```

Comment:
Why does the matmul need to be aware of padded inputs? If input is padded, output should also be padded.

Reply:
the in1 reader needs it, so it doesn't send back un-used data to storage cores